### PR TITLE
chore: Add compound index to user class

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Add index to user class for better performance. 

Fixes https://theappsmith.slack.com/archives/C01EDBBC4JW/p1709925172694189

## Automation

/ok-to-test tags="tag.all"

### :mag: Cypress test results

> [!CAUTION]  
> If you modify the content in this section, you are likely to disrupt the CI result for your PR.

## Summary by CodeRabbit

- **Performance Improvements**
	- Improved query performance by adding a new index to the User database collection.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @RollbackExecution
+    public void rollbackExecution() {}
+
+    @Execution
+    public void addMissingIndexInUserCollection() {
+
+        Index userEmailCreatedAtIndex =
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java:30]: Let's add a drop if exists for retries please

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                makeIndex("deleted", "deletedAt", "email", "createdAt").unique().named("user_compound_index");
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java:31]: Let's name the index to reflect its meaning like how we usually do as well

Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java:31]: Can you also check if what we see in the prod DB is reflected exactly here? There was a sort criteria there that seems to be missing.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import io.mongock.api.annotations.RollbackExecution;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.index.Index;
+
+import static com.appsmith.server.migrations.DatabaseChangelog1.ensureIndexes;
+import static com.appsmith.server.migrations.DatabaseChangelog1.makeIndex;
+
+@Slf4j
+@ChangeUnit(order = "047", id = "add-missing-index-in-user-collection", author = " ")
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java:15]: Nit: Order and file name are out of sync

**Review by nidhi-nair — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java (hunk 2 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration048AddIndexToUserCollection.java:31]: I have checked this and i don't see any criteria for the sort.

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**